src/util/object_factory_parameters.cpp

```diff
@@ -46,6 +50,34 @@ void object_factory_parameterst::set(const optionst &options)
     min_nondet_string_length =
       options.get_unsigned_int_option("min-nondet-string-length");
   }
+  if(options.is_set("java-assume-inputs-interval"))
+  {
+    const auto &interval = options.get_option("java-assume-inputs-interval");
+    const std::regex limits_regex("\\[(-\\d+|\\d*):(-\\d+|\\d*)\\]");
+    std::smatch base_match;
+    if(!std::regex_match(interval, base_match, limits_regex))
+    {
+      throw invalid_command_line_argument_exceptiont(
+        "interval must be of the form [int:int], [int:] or [:int]",
+        "--java-assume-inputs-interval");
+    }
+    assume_inputs_interval = [&]() -> integer_intervalt {
+      integer_intervalt temp;
+      if(base_match[1] != "")
```

Comment:
Why not `!base_match[1].empty()`

Reply:
The type doesn't have `empty()` function. Note that because the regex allows empty string here, this would always be a match. I just need to check that what matched was an empty string (no bound).